src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1090,15 +1092,16 @@ public int OnAfterCloseSolution(object pUnkReserved)
 
         private bool _disposed = false;
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD010:Invoke single-threaded types on Main thread", Justification = "https://github.com/NuGet/Home/issues/10933")]
         public void Dispose()
         {
             if (!_disposed)
             {
                 _disposed = true;
 
-                if (ThreadHelper.JoinableTaskContext?.IsOnMainThread == true)
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
nit: Async disposable might be an option. I understand if you resolve this as out of scope.

Reply:
I did a quick test, and the dispose method wasn't called when `IAsyncDisablable`. Maybe VS-MEF doesn't support `IAsyncDisposable`?